packages/calypso-e2e/src/lib/flows/gutenboarding-flow.ts

```diff
@@ -0,0 +1,205 @@
+import { Page } from 'playwright';
+import { getViewportName } from '../../browser-helper';
+
+export type Plans = 'Free' | 'Personal' | 'Premium' | 'Business' | 'eCommerce';
+export type Features =
+	| 'Custom domains'
+	| 'Store'
+	| 'SEO tools'
+	| 'Plugins'
+	| 'Ad-free'
+	| 'Image storage'
+	| 'Video storage'
+	| 'Priority support';
+
+const selectors = {
+	button: ( text: string ) => `button:text("${ text }")`,
+
+	// Start your website
+	siteTitle: '.acquire-intent-text-input__input',
+
+	// Domain
+	domainSearch: 'input[placeholder="Search for a domain"]',
+	domainSuggestionSpan: ( name: string, tld: string ) =>
+		`.domain-picker__suggestion-item-name span:has-text("${ name }${ tld }")`,
+
+	// Design
+	design: ( name: string ) => `button[data-e2e-button="freeOption"] span:text("${ name }")`,
+	fontPairingButton: ( platform: 'mobile' | 'desktop', fontName: string ) => {
+		return `.style-preview__font-options-${ platform } span:text("${ fontName }")`;
+	},
+	mobileFontPairingDropdown: `button.style-preview__font-option-select`,
+
+	// Features
+	featureItem: `button.features__item`,
+
+	// Plans
+	planItem: '.plans-accordion-item',
+	planName: '.plans-accordion-item__name',
+};
+
+/**
+ * Implements actions of the Gutenboarding process.
+ */
+export class GutenboardingFlow {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Given a text, clicks on the first instance of the button with the text.
+	 *
+	 * @param {string} text User-visible text on the button.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async clickButton( text: string ) {
+		await this.page.click( selectors.button( text ) );
```

Comment:
A generic `clickButton` method to handle clicking of any buttons that can be located uniquely on the page.

Reply:
Kinda going back and forth on this one personally.

This was never codified in the documentation but IIRC @WunderBart and I discussed this and it was agreed to avoid what had befallen the Selenium e2e suite (lots of helper functions, some redundant, some not used, some not even working properly). Do you recall anything of the sort, @WunderBart?

I'll reveal my hand first; I would prefer to implement these small helper functions individually in the POMs. Reason being:
1. normally we work with pages, but sometimes with frames;
2. 'local' implementation will be better suited to the subtle differences present on that page;

Case in point, in another migration PR I implemented another `clickButton` method but with intricacies specific to that page, which wouldn't have worked with how `Gutenboarding` pages work.

Reply:
> avoid what had befallen the Selenium e2e suite (lots of helper functions, some redundant, some not used, some not even working properly)

This makes a lot of sense to me!

> My opinion is that it is preferable to implement these small helper functions individually in the POMs, at risk of repetitiveness.

Totally down for this. I think it fits how we're trying to grow well!

In that case, should we make this method private?

Reply:
It should not be private as it's used by the test spec to perform actions on the page. I don't think it was valuable creating a method for each individual button action (eg. clickContinue, clickCancel, clickXYZ) as it only clutters the class. Do you think that's preferable for clarity?

Reply:
Ah whoops, I missed that, sorry! Hmm, no, I've actually really come around on this. I think this is good as is!